### PR TITLE
temporarily block login on podspaces, display warning

### PR DESCRIPTION
**Pull request by VirginiaBalseiro**
## Description

Temporarily block login on Podspaces until migration is completed

## Commit checklist

- [x] All acceptance criteria are met.
- [x] Includes tests to ensure functionality and prevent regressions.
- [ ] Relevant documentation, if any, has been written/updated.
- [ ] The changelog has been updated, if applicable.

### CODE REVIEW

**Review by chelseapinka — APPROVED** [review]
Makes sense and looks nice to have message appear on Signin button click.

File: components/login/index.jsx [hunk 1]
```diff
@@ -45,12 +45,21 @@ const useStyles = makeStyles((theme) => createStyles(styles(theme)));
...
 
   const login = useCallback(
     (oidcIssuer) => {
+      if (
+        oidcIssuer.includes("login.inrupt.com") ||
+        oidcIssuer.includes("broker.pod.inrupt.com")
+      ) {
+        setShouldDisplayWarning(true);
+        return;
+      }
```
Comment (ThisIsMissEm) [inline components/login/index.jsx:62]:
```suggestion
      } else {
         setShouldDisplayWarning(false);
      }
```

File: components/login/index.test.jsx [hunk 2]
```diff
@@ -118,4 +119,27 @@ describe("Login form", () => {
...
+
+    act(() => {
+      signinButton.click();
+    });
+
+    expect(mockUseReturnUrl.persist).not.toHaveBeenCalled();
+    expect(session.login).not.toHaveBeenCalled();
+
+    expect(asFragment).toMatchSnapshot();
+  });
```
Comment (ThisIsMissEm) [inline components/login/index.test.jsx:144]: Maybe add a test case proving that CSS/NSS could be used? idk.

File: components/login/index.jsx (hunk 1 above)
Comment (VirginiaBalseiro) [reply components/login/index.jsx:62]: If `shouldDisplayWarning` is set to false by default, and only changed to true if this condition is true, do we need this?

Comment (ThisIsMissEm) [reply components/login/index.jsx:62]: Well, you can type in one idp, then get the warning, then type in a different one

Comment (VirginiaBalseiro) [reply components/login/index.jsx:62]: but if that happens you're redirected, as this is the login function and it is not called on value change, so the warning will stay there even when typing and won't clear until it is submitted

File: components/login/index.test.jsx (hunk 2 above)
Comment (VirginiaBalseiro) [reply components/login/index.test.jsx:144]: Added test - passed an IDP value via query params to save myself the whole clicking through the dropdown thing.

**Review by ThisIsMissEm — APPROVED** [review]
Looks good to me.

**Comment by ThisIsMissEm** [comment]
wait.

File: components/login/index.jsx [hunk 3]
```diff
@@ -79,6 +88,15 @@ export default function LoginForm() {
 
   return (
     <div className={bem("login-form")}>
+      {/* FIXME: remove after pod migration is completed */}
+      {shouldDisplayWarning && (
+        <Message variant="warning" prominent hasIcon>
+          Sorry, we cannot log you in at the moment due to planned system
+          maintenance. Please see our
+          <a href="https://inrupt.com/blog/pod-spaces-upgrade"> blog post </a>
+          for more information.
```
Comment (ThisIsMissEm) [inline components/login/index.jsx:97]:
```suggestion
          maintenance. Please see our{" "}
          <a href="https://inrupt.com/blog/pod-spaces-upgrade">blog post</a>{" "}
          for more information.
```

To fix the link underline:
Screenshot 2022-07-20 at 17 22 52

**Review by ThisIsMissEm — CHANGES_REQUESTED** [review]

**Review by ThisIsMissEm — APPROVED** [review]

**Pull request merged**